### PR TITLE
Some cleanup and added PSR7 as requirement

### PR DESCRIPTION
**Pull request by Nyholm**
I copied from the specification document.

### CODE REVIEW

File: src/ClientInterface.php [hunk 1]
```diff
@@ -10,7 +10,7 @@
     /**
      * Sends a PSR-7 request and returns a PSR-7 response.
      *
-     * Every technically correct HTTP response MUST be returned as is, even if it represents an HTTP
+     * Every technically correct HTTP response MUST be returned as is, even if it represents a HTTP
```
Comment (sagikazarmark) [inline src/ClientInterface.php:13]: You sure?

https://www.quora.com/A-http-or-an-http-which-is-the-correct-grammatical-usage

Comment (Nyholm) [reply src/ClientInterface.php:13]: Im sure I **copied** it right, but I think I agree with you on the grammar. 

https://github.com/php-fig/fig-standards/blob/master/proposed/http-client/http-client.md#clientinterface

I do not think it is a blocker for the package. Could you please correct this at the specification?

Comment (sagikazarmark) [reply src/ClientInterface.php:13]: Done #1006

**Review by sagikazarmark — APPROVED** [review]

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**